### DISCUSSION TITLE
Upgrade  of Firebase V1 authentication for a legacy application

### DISCUSSION THREAD

**Discussion by swapna15** [kreait/firebase-php#955] [General]
Hi,

I am using Laravel application which is a very old legacy application and needs the v1 auth for firebase to be integrated. As the application is audio or video calls most of the application is dependent on this firebase event. When I try to install the kreait/firebase-php facing issue with composer.json file getting updated by not the composer.lock. I think its a php version or Laravel version issue but if you can help me know which version of the app supports this basic authentication it would be helpful.

**Comment by jeromegamez** [comment]
[The library should support Laravel 9.0 and up](https://github.com/kreait/laravel-firebase/blob/95bf4ec181b27cb415c1af9788f4b076f8e127a5/composer.json#L21). If it doesn't work, you can check with the [`composer why-not`](https://getcomposer.org/doc/03-cli.md#prohibits-why-not) command.
